src/manage_nsfs/manage_nsfs_glacier.js

```diff
@@ -69,12 +69,38 @@ async function process_expiry() {
     const fs_context = native_fs_utils.get_process_fs_context();
 
     await lock_and_run(fs_context, SCAN_LOCK, async () => {
-        if (!(await time_exceeded(fs_context, config.NSFS_GLACIER_EXPIRY_INTERVAL, GlacierBackend.EXPIRY_TIMESTAMP_FILE))) return;
+        const backend = getGlacierBackend();
+        if (
+            await backend.low_free_space() ||
+            is_desired_time(new Date(), config.NSFS_GLACIER_EXPIRY_RUN_TIME)
+        ) {
+            await backend.expiry(fs_context);
+        }
+    });
+}
 
+/**
+ * is_desired_time returns true if the given time matches with
+ * the desired time
+ * @param {Date} current
+ * @param {string} desire time in format 'hh:mm'
+ * @returns {Boolean}
+ */
+function is_desired_time(current, desire) {
+    const current_hour = current.getHours();
+    const current_min = current.getMinutes();
+
+    const [desired_hour, desired_min] = desire.split(':').map(Number);
+    if (
+        isNaN(desired_hour) ||
+        isNaN(desired_min) ||
+        (desired_hour < 0 || desired_hour >= 24) ||
+        (desired_min < 0 || desired_min >= 60)
+    ) {
+        throw new Error('invalid `NSFS_GLACIER_EXPIRY_RUN_TIME` - must be hh:mm');
+    }
 
-        await getGlacierBackend().expiry(fs_context);
-        await record_current_time(fs_context, GlacierBackend.EXPIRY_TIMESTAMP_FILE);
-    });
+    return desired_hour === current_hour && desired_min === current_min;
```

Comment:
So in case we are not scheduled to run during the desired minute, we miss the daily opportunity completely?
Perhaps we should be a bit more tolerant and have a maximum interval after that desired time where we would still run the expiry? In order to do that we still need to maintain the timestamp file to prevent running it multiple times in that window.